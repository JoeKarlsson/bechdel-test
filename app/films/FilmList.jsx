import React from 'react';
import FilmItem from './FilmItem.jsx';
import styles from './FilmList.scss';

class FilmList extends React.Component {
  render() {
    let filmListNode = this.props.films.map((filmData) => {
      return (
        <FilmItem
          film={filmData}
          key={filmData._id}
          className={styles.filmListNode}
        />
      );
    });

    return (
      <div className={styles.FilmList}>
        {filmListNode}
      </div>
    );
  }
}

FilmList.propTypes = {
  films: React.PropTypes.array,
};

FilmList.defaultProps = {
  films: [],
};

export default FilmList;
