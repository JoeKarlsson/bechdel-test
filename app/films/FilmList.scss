.filmList {
  display: flex;
  flex-wrap: wrap;
}

.filmListNode {
}
