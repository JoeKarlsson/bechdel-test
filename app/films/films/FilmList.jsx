import React from 'react';
import FilmItem from './FilmItem.jsx';
import styles from './FilmList.scss';

class FilmList extends React.Component {
  render() {
    let filmListNode = this.props.films.map((filmData) => {
      return (
        <FilmItem
          film={filmData}
          key={filmData._id}
          className='filmListNode'
        />
      );
    });

    return (
      <div className='filmList'>
        {filmListNode}
      </div>
    );
  }
}

FilmList.propTypes = {
  films: React.PropTypes.array,
};

FilmList.defaultProps = {
  films: [],
};

export default FilmList;
