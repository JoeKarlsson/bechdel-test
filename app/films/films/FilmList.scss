.filmList {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}

.filmListNode {
}
