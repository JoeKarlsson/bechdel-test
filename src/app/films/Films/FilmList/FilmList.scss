.FilmList {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-around;
}
