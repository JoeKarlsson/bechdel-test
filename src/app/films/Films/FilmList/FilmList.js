import React from 'react';
import PropTypes from 'prop-types';
import FilmItem from '../FilmItem/FilmItem';
import './FilmList.scss';

const FilmList = props => {
	console.log(props.films.length);
	if (props.films.length === 0) {
		return <div>No Films Have Been Added Yet</div>;
	}
	const filmListNode = props.films.map(filmData => {
		return (
			<FilmItem film={filmData} key={filmData._id} className="filmListNode" />
		);
	});

	const numFillerNodes = props.films.length % 6;

	const arr = new Array(numFillerNodes);
	const fillerNode = arr.map(() => {
		return <div className="fillerNode" />;
	});

	return (
		<div className="FilmList">
			{filmListNode}
			{fillerNode}
		</div>
	);
};

FilmList.propTypes = {
	films: PropTypes.array, // eslint-disable-line react/forbid-prop-types
};

FilmList.defaultProps = {
	films: [],
};

export default FilmList;
