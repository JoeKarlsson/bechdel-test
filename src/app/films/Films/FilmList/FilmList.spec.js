import React from 'react';
import {
	shallow,
	configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import FilmList from './FilmList';

configure({ adapter: new Adapter() });

describe('FilmList', () => {
	let wrapper;

	global.requestAnimationFrame = (callback) => {
		setTimeout(callback, 0);
	};

	beforeEach(() => {
		wrapper = shallow(<FilmList />);
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('is rendered', () => {
				const component = renderer.create(<FilmList />);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});

			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});

			it('should be selectable by the class `FilmList`', () => {
				expect(wrapper.is('.FilmList')).toBe(true);
			});

			it('should mount in the full DOM', () => {
				expect(wrapper.find('.FilmList').length).toBe(1);
			});

			it('should have correct inital instance', () => {
				const initialInstance = wrapper.instance();
				const expectedInstance = null;
				expect(initialInstance).toBe(expectedInstance);
			});
		});
	});
});
