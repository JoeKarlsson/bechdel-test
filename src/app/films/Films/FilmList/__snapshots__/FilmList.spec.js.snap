// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList rendering initial state is rendered 1`] = `
<div>
  No Films Have Been Added Yet
</div>
`;
